pkg/pluginloader/loader.go

```diff
@@ -0,0 +1,84 @@
+package pluginloader
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"os/exec"
+	"strconv"
+
+	"github.com/spf13/cobra"
+)
+
+// TODO:
+// Unmarshal the YAML
+// Extract the commands using plugin.go (command structure)
+
+// Create a Cobra Command
+func createCommand(cmdStruct *CommandConfig) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     cmdStruct.Name,
+		Short:   cmdStruct.ShortDescription,

```

Comment:
Let's review what properties of command we want to expose later (separate issue) - there is validation etc.